system/src/system_network_internal.h

```diff
@@ -599,34 +613,49 @@ class ManagedNetworkInterface : public NetworkInterface
         return (SPARK_WLAN_STARTED && WLAN_CONNECTING);
     }
 
-    void on() override
+    int on() override
     {
         LOG_NETWORK_STATE();
-        if (!SPARK_WLAN_STARTED)
+        if (!SPARK_WLAN_STARTED || !WLAN_INITIALIZED)
         {
+            bool startedPowerOn = !SPARK_WLAN_STARTED;
+            WLAN_INITIALIZED = 0;
             const auto diag = NetworkDiagnostics::instance();
             diag->status(NetworkDiagnostics::TURNING_ON);
-            system_notify_event(network_status, network_status_powering_on);
+            if (startedPowerOn) {
+                // Suppress these events
+                system_notify_event(network_status, network_status_powering_on);
+            }
             config_clear();
             const int ret = on_now();
             if (ret != 0) {
                 diag->lastError(ret);
+                if (!wlan_watchdog_duration) {
+                    // Just in case arm WLAN WD
+                    ARM_WLAN_WD(POWER_ON_WD);
+                }
```

Comment:
I think this should at a minimum be a 10 minute timer due to auto COPS and network compliance.

Reply:
Right... it's safety code.  But if it ever comes into effect, let's default to 10 minutes... or just remove it.

Reply:
Any reason we can't keep this at 5 minutes? 10 minutes seems like an awfully long time to wait for the modem to power on, initialize the SIM and perform some additional initialization?

Reply:
Since it's a timeout we should never hit, I was thinking we should stay on the safe side of cellular network compliance.  If we do run into it continuously for whatever odd reason, at least we won't be doing anything potentially harmful.  If it would block a user from short circuiting recovery on their own though, 5 minutes might even be too long.

Reply:
Changed to 10